lib/archethic/mining/pending_transaction_validation.ex

```diff
@@ -275,6 +175,64 @@ defmodule Archethic.Mining.PendingTransactionValidation do
     end
   end
 
+  defp validate_contract(%Transaction{data: %TransactionData{code: code}})
+       when byte_size(code) > @code_max_size do
+    {:error, "invalid contract type transaction , code exceed max size"}
+  end
+
+  @spec validate_ownerships(Transaction.t()) :: :ok | {:error, any()}
+  defp validate_ownerships(%Transaction{data: %TransactionData{ownerships: ownerships}}) do
```

Comment:
You can use pattern matching to simplify the checks. 
The `lenght == 0` can be simplify over pattern matching: ` ownerships : []`

So the do_validate_ownerships can be removed and moved into the validate_ownerships